Datadog/Example/Debugging/BackgroundEvents/BackgroundLocationMonitor.swift

```diff
@@ -49,6 +49,10 @@ internal class BackgroundLocationMonitor: NSObject, CLLocationManagerDelegate {
         return UIApplication.shared.applicationState == .background
     }
 
+    private var rum: DDRUMMonitor {
+        RUMMonitor.shared()
+    }
```

Comment:
It's worth considering setting this via default init param, just to cut all the direct singleton dependency away.

```
init(rum: DDRUMMonitor = .shared()) {
  self.rum = rum
}
```